src/main/java/com/uid2/shared/store/ScopedStoreReader.java

```diff
@@ -0,0 +1,69 @@
+package com.uid2.shared.store;
+
+import com.uid2.shared.Utils;
+import com.uid2.shared.attest.UidCoreClient;
+import com.uid2.shared.cloud.CloudStorageException;
+import com.uid2.shared.cloud.ICloudStorage;
+import com.uid2.shared.store.parser.ParsingResult;
+import com.uid2.shared.store.parser.Parser;
+import com.uid2.shared.store.scope.StoreScope;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ScopedStoreReader<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScopedStoreReader.class);
+    private final ICloudStorage metadataStreamProvider;
+    private final StoreScope scope;
+    private final Parser<T> parser;
+    private final String dataTypeName;
+    private final ICloudStorage contentStreamProvider;
+    private final AtomicReference<T> latestSnapshot = new AtomicReference<T>();
```

Comment:
You can remove the `T` in `new`
i.e. `private final AtomicReference<T> latestSnapshot = new AtomicReference<>();`

Reply:
ok